### PR TITLE
Add new ascsers_cluster gatherer

### PR DESCRIPTION
**Pull request by arbulu89**
New `ascsers_cluster` gatherer, which gathers details about this type of clusters. **Open for other name options**
The gatherer doesn't require any argument.

To use, simply:
```yaml
facts:
  - name: cluster_details
    gatherer: ascsers_cluster
```

This is what it returns (once approved, this will be commented in the gatherers docs):

```ts
#{
  "PRD": #{
    "ensa_version": "ensa1/ensa2/unknown",
    "instances": [
      #{
         "name" : "ASCS00", // 2nd value of the InstanceName value
         "instance_number: "00", 
         "virtual_hostname": "sapascs00", // 3rd value of the InstanceName value
         "filesystem_based": true/false, // if the instance resource group has a FileSystem resource
         "local": true/false // if the instance is running locally (sapcontrol returns a positive response for GetProcessList)
      },...
   ]
  },
  "DEV": #{
     "ensa_version": "ensa1/ensa2/unknown",
     "instances": [
...
  }
}
```

FYI: @yeoldegrove

### CODE REVIEW

**Review by nelsonkopliku — APPROVED** [review]
LGTM, I guess :smile: 

With regards to naming, I often captured `managed_systems` in the discussions about this.

What about something like:
- `ascsers_cluster_managed_systems`
- `ascsers_managed_systems`
- `cluster_managed_systems`

Great job!

**Review by EMaksy — APPROVED** [review]
Hey from my side looks good :+1:

**Comment by arbulu89** [comment]
@yeoldegrove thank you so much!
the changes look good to me. I'm waiting for some additional review from my team, but i think we can merge it the next days for sure

**Review by nelsonkopliku — APPROVED** [review]
LGTM

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (CDimonaco) [inline internal/factsengine/gatherers/ascsers_cluster.go:217]: This could panic, because in the second or condition err if nil and it's used in logger

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (CDimonaco) [inline internal/factsengine/gatherers/ascsers_cluster.go:223]: Could we add a fallthrough switch for this checks?

**Review by CDimonaco — COMMENTED** [review]
Good job man, I left some comments

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (arbulu89) [reply internal/factsengine/gatherers/ascsers_cluster.go:223]: How would you use it?
I don't see where I could use the `fallthrough`.
I could write it as a switch like this, but it doesn't look much better:
```

switch process.Name {
case "enserver":
	return Ensa1, true
case "enrepserver":
	return Ensa1, true
case "enq_server":
	return Ensa2, true
case "enq_replicator":
	return Ensa2, true
}
		}
```

Comment (CDimonaco) [reply internal/factsengine/gatherers/ascsers_cluster.go:223]: Yeah Sorry, I used the wrong words, I mean we could use multiple values for the case statement,  it's evaluated like an or condition

**Review by CDimonaco — APPROVED** [review]

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (arbulu89) [reply internal/factsengine/gatherers/ascsers_cluster.go:155]: Done

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (arbulu89) [reply internal/factsengine/gatherers/ascsers_cluster.go]: Done

File: internal/factsengine/gatherers/ascsers_cluster.go
Comment (arbulu89) [reply internal/factsengine/gatherers/ascsers_cluster.go]: Done

**Pull request merged**
